amlb/results.py

```diff
@@ -684,71 +684,108 @@ def r2(self):
         """R^2"""
         return float(r2_score(self.truth, self.predictions))
 
-class TimeSeriesResult(RegressionResult):
 
+class TimeSeriesResult(RegressionResult):
     def __init__(self, predictions_df, info=None):
         super().__init__(predictions_df, info)
-        self.truth = self.df['truth'].values if self.df is not None else None #.iloc[:, 1].values if self.df is not None else None
-        self.predictions = self.df['predictions'].values if self.df is not None else None #.iloc[:, -2].values if self.df is not None else None
-        self.y_past_period_error = self.df['y_past_period_error'].values
-        self.quantiles = self.df.iloc[:, 2:-1].values
-        self.quantiles_probs = np.array([float(q) for q in self.df.columns[2:-1]])
-        self.truth = self.truth.astype(float, copy=False)
-        self.predictions = self.predictions.astype(float, copy=False)
-        self.quantiles = self.quantiles.astype(float, copy=False)
-        self.y_past_period_error = self.y_past_period_error.astype(float, copy=False)
+        required_columns = {'truth', 'predictions', 'repeated_item_id', 'repeated_abs_seasonal_error'}
+        if required_columns - set(self.df.columns):
+            raise ValueError(f'Missing columns for calculating time series metrics: {required_columns - set(self.df.columns)}.')
+
+        quantile_columns = [column for column in self.df.columns if column.startswith('0.')]
+        unrecognized_columns = [column for column in self.df.columns if column not in required_columns and column not in quantile_columns]
+        if len(unrecognized_columns) > 0:
+            raise ValueError(f'Predictions contain unrecognized columns: {unrecognized_columns}.')
 
-        self.target = Feature(0, 'target', 'real', is_target=True)
         self.type = DatasetType.timeseries
+        self.truth = self.df['truth'].values.astype(float)
+        self.item_ids = self.df['repeated_item_id'].values
+        self.abs_seasonal_error = self.df['repeated_abs_seasonal_error'].values.astype(float)
+        # predictions = point forecast, quantile_predictions = quantile forecast
+        self.predictions = self.df['predictions'].values.astype(float)
+        self.quantile_predictions = self.df[quantile_columns].values.astype(float)
+        self.quantile_levels = np.array(quantile_columns, dtype=float)
 
-    @metric(higher_is_better=False)
-    def mase(self):
-        """Mean Absolute Scaled Error"""
-        return float(np.nanmean(np.abs(self.truth/self.y_past_period_error - self.predictions/self.y_past_period_error)))
+        if (~np.isfinite(self.predictions)).any() or (~np.isfinite(self.quantile_predictions)).any():
+            raise ValueError('Predictions contain NaN or Inf values')
+
+        _, unique_item_ids_counts = np.unique(self.item_ids, return_counts=True)
+        if len(set(unique_item_ids_counts)) != 1:
+            raise ValueError(f'Error: Predicted sequences have different lengths {unique_item_ids_counts}.')
```

Comment:
Isn't the more correct test that each predicted sequence should be of length `truth`?